src/kudu/consensus/quorum_util.cc

```diff
@@ -922,10 +945,61 @@ void AdjustVoterDistributionWithCurrentVoters(
   }
 }
 
+void GetVoterDistributionForQuorumId(
+    const RaftConfigPB& config,
+    std::map<std::string, int>* quorum_id_vd
+) {
+  if (IsUseQuorumId(config.commit_rule())) {
+    // Step 1: Using the default quorum size
+    for (const auto peer : config.peers()) {
+      if (peer.has_member_type() && peer.member_type() == RaftPeerPB::VOTER) {
```

Comment:
nit: using continue pattern is easier to reason about and saves extra levels of indentation.
(!peer.has_member_type() || peer.member_type() != VOTER)
   continue;